src/types.ts

```diff
@@ -1,10 +1,16 @@
 import { BigNumberish } from "@ethersproject/bignumber";
 import { BytesLike } from "@ethersproject/bytes";
 
+export enum AllocationType {
+  simple,
+  withdrawHelper,
+  guarantee
+}
+
 export interface Allocation {
   destination: string; // an Ethereum address
   amount: BigNumberish;
-  callTo: string; // an Ethereum address
+  allocationType: BigNumberish;
```

Comment:
I think maybe we can use a javascript `number`, here? That should be safe if we are only dealing with a `uint8`?